code/range_overlap.m

```diff
@@ -16,3 +16,31 @@
     output_range = [lowest, highest];
 
 end
+
+% Solution to the exercise
+%
+% function output_range = range_overlap(varargin)
+% 
+% % Return common overlap among a set of [low, high] ranges.
+% 
+% lowest = -inf;
+% highest = +inf;
+% 
+% for i = 1:nargin
+%     range = varargin{i};
+%     low = range(1);
+%     high = range(2);
+%     lowest = max(lowest, low);
+%     highest = min(highest, high);
+%     
+%     % catch non-overlapping ranges
+%     if low >= highest || high<=lowest
+%         output_range=NaN;
+%         return
+%     end
+% end
+% 
+% output_range = [lowest, highest];
+% 
+% end
+
```

Comment:
I'll comment directly when you raise an issue on varargin, but I think it might be a good "exercise for the reader" to look it up in the help, after a brief explanation from the tutor.